### PR TITLE
VZ-5537: Add install/upgrade/uninstall of prometheus-adapter

### PR DESCRIPTION
**Pull request by fredtibbitts**
# Description

This PR adds support for installing, upgrading, and uninstalling the Prometheus Adapter. 

The helm chart was added in a previous PR, as were the updates to the Verrazzano custom resource to support the new component. A follow-on PR will add some basic acceptance tests to validate the component lifecycle.

Implements VZ-5537

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/verrazzano-bom.json [hunk 1]
```diff
@@ -435,6 +435,18 @@
               "helmTagKey": "prometheusOperator.image.tag"
             }
           ]
+        },
+        {
+          "repository": "verrazzano",
```
Comment (davisgomes) [inline platform-operator/verrazzano-bom.json:440]: Do we want all of the new monitoring components as subcomponents to the prometheus-operator?

**Review by davisgomes — DISMISSED** [review]

**Review by stevshap — APPROVED** [review]

**Review by desagar — APPROVED** [review]

File: platform-operator/verrazzano-bom.json (hunk 1 above)
Comment (pfmackin) [reply platform-operator/verrazzano-bom.json:440]: Looks like an independent component from the VZ CR

**Review by pfmackin — DISMISSED** [review]

File: platform-operator/verrazzano-bom.json (hunk 1 above)
Comment (fredtibbitts) [reply platform-operator/verrazzano-bom.json:440]: I'll move it.

Comment (fredtibbitts) [reply platform-operator/verrazzano-bom.json:445]: Done.

**Review by desagar — DISMISSED** [review]

**Review by davisgomes — APPROVED** [review]

**Review by stevshap — APPROVED** [review]

**Pull request merged**